### ISSUE TITLE
Module Generator: Class 'insolita\wgadminlte\SmallBox' not found

### ISSUE THREAD

**Issue by LarryTX** [schmunk42/yii2-giiant#248]
I used your Module Generator, especially because I like what it provides when the user goes to the module directory without specifying any model or controller. One observation: It would be really, really, really helpfull if you could specify any dependencies in your instructions, when those dependencies are not installed as dependencies when your extension is installed. What happens now is that the extension is installed with no notice, whatsoever, that many required dependencies are missing and will be show-stoppers periodically and without notice, sometimes seeming to come for no reason at all. In this case, I've been using giiant for a very long time, including your module generator, without issues arising. Then todya, I decides to work with going to /<module_name> as opposed to /<module_name>/<controller_route>, and suddenly out of the blue, there are seemingly dozens of required extensions that need to be added to even get past the error messages. That means that I've got to evaluate each and every single required extension to see if I'm even willing to install them in a Yii2 project, then if I'm willing, install them, only to find that they require other extensions. What was going to be a 30 minute task suddenly balloons into a full day or two or three.

I had already discovered with your module generator that I immediately need to disable the dmstr\web\traits\AccessBehaviorTrait because I have absolutely no need for it as I do security and authorization in an entirely different way. Or in your CRUD generator,, I have to through and immediately replace use \dmstr\bootstrap\Tabs; with kartik\tabs\TabsX because you don;'t offer alternatives and because Kartik offers such a complete  and consistent solution (even if he is so very difficult to deal with).

Nonetheless, the current situation doesn't help all that. First, in trying to go /<module_name>, you are confronted with an error because you need to have rmrevin/yii2-fontawesome installed. No, I don't need that extension. I load the FontAwesome 5 Pro set via CSS already. But I didn't have time to figue out a workaround, so I installed it. The, the next thing is I get a Class 'insolita\wgadminlte\SmallBox' not found error message. I actually have customized AdminLTE 3 (Bootstrap 4) as a backend layout, so this one was interesting for me. So I installed insolita/yii2-adminlte-widgets.

Did that fix is? No, of course not. There is no SmallBox class in insolita/yii2-adminlte-widgets. There is a LteSmallBox class in the insolita/yii2-adminlte-widgets that I installed. Now, I have to ask, is there any end to this?

Do you have any workaround for this? anyway to get around this show-stopper and keep moving. Or should I just ditch the module development portion of giiant and do it on my own? (I hate to think about how many more extensions I would have to work through and fix just to get a workable solution.)

**Comment by schmunk42** [comment]
Yeah it's true that the default templates are tightly integrated with phd. Would be happy to sort those issues out - and also add it to the docs.

But nonetheless, defaults should be changed in a way, so that there are minimal external requirements.

Actually the `crud\Generator` has a property `requires` which displays requirements on CLI. But it's currently only filled by providers.

But you can also use other templates (see eg. `crud/editable/views`).

About `dmstr\bootstrap\Tabs` - we created that because there way no other components which could remember the active tab back then. If there's a better one, I'd happy to replace it.
